pkg/kclient/fake/discovery.go

```diff
@@ -0,0 +1,63 @@
+package fake
+
+import (
+	"fmt"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/discovery/fake"
+	"runtime"
+	"sync"
+)
+
+type ResourceMapEntry struct {
+	list *metav1.APIResourceList
+	err  error
+}
+
+type FakedDiscovery struct {
+	*fake.FakeDiscovery
+
+	lock        sync.Mutex
+	resourceMap map[string]*ResourceMapEntry
+}
+
+func NewFakeDiscovery() *FakedDiscovery {
+	return &FakedDiscovery{resourceMap: make(map[string]*ResourceMapEntry, 7)}
+}
+
+func (c *FakedDiscovery) AddResourceList(key string, are *metav1.APIResourceList) {
+	c.resourceMap[key] = &ResourceMapEntry{list: are}
+}
+
+func (c *FakedDiscovery) ServerResourcesForGroupVersion(groupVersion string) (*metav1.APIResourceList, error) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if rl, ok := c.resourceMap[groupVersion]; ok {
+		return rl.list, rl.err
+	}
+	return nil, kerrors.NewNotFound(schema.GroupResource{}, "")
+}
+
+func (c *FakedDiscovery) ServerVersion() (*version.Info, error) {
```

Comment:
This is already in https://github.com/openshift/odo/blob/89ac67e425bffe8c400645bca933d687b5ac8b19/pkg/occlient/fakeclient.go#L132 shouldn't it be reused?

Reply:
I moved that code from occlient to kclient, I should remove the occlient side of it